src/impl.cpp

```diff
@@ -1 +1,22 @@
 #include "impl.h"
+#include <iostream>
+#include <istream>
+#include <vector>
+#include <iterator>
+#include <sstream>
+#include <algorithm>
+#include <unordered_set>
+
+std::vector<std::string> uniqueWord(const std::string string)
+{
+    std::string text = std::move(string);
+ 
+    std::transform(text.begin(), text.end(), text.begin(), [](unsigned char c){ return std::tolower(c); });
+    std::istringstream iss(text);
+    std::vector<std::string> results((std::istream_iterator<std::string>(iss)), std::istream_iterator<std::string>());
+    std::unordered_set<std::string> set(std::move(results.begin()), std::move(results.end()));
```

Comment:
:)
please read about move iterators and it's generator method
https://en.cppreference.com/w/cpp/iterator/make_move_iterator